src/rgw/rgw_sal_motr.cc

```diff
@@ -1246,8 +1246,38 @@ MotrObject::~MotrObject() {
 
 int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx, Attrs* setattrs, Attrs* delattrs, optional_yield y, rgw_obj* target_obj)
 {
-  // TODO: implement
-  ldpp_dout(dpp, 20) <<__func__<< ": MotrObject::set_obj_attrs()" << dendl;
+  if (this->category == RGWObjCategory::MultiMeta)
+    return 0;
+
+  string bname, key;
+  if (target_obj) {
+    bname = get_bucket_name(target_obj->bucket.tenant, target_obj->bucket.name);
+    key   = target_obj->key.to_str();
+  } else {
+    bname = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
+    key   = this->get_key().to_str();
+  }
+  ldpp_dout(dpp, 20) << "MotrObject::set_obj_attrs(): "
+                    << bname << "/" << key << dendl;
+
+  // Encode object's metadata (those stored in rgw_bucket_dir_entry).
+  bufferlist bl;
+  rgw_bucket_dir_entry ent;
+  ent.encode(bl);
+  encode(attrs, bl);
+
+  if (this->store->get_obj_meta_cache()->get(dpp, key, bl)) {
```

Comment:
PUT operation is needed to update data in backend store, plus cache also later. This is not GET where if you get data from cache, can return data from there itself.

Reply:
done

Reply:
Done